lib/redlock/client.rb

```diff
@@ -262,17 +273,29 @@ def lock_instances(resource, ttl, options)
       value = (options[:extend] || { value: SecureRandom.uuid })[:value]
       allow_new_lock = options[:extend_only_if_locked] ? 'no' : 'yes'
 
+      errors = []
+
       locked, time_elapsed = timed do
-        @servers.select { |s| s.lock resource, value, ttl, allow_new_lock }.size
-      end
+        @servers.select do |s|
+          s.lock(resource, value, ttl, allow_new_lock)
+        rescue RedisClient::ConnectionError => e
+          errors << e
+          next
+        end.size
+      end 
 
       validity = ttl - time_elapsed - drift(ttl)
 
       if locked >= @quorum && validity >= 0
         { validity: validity, resource: resource, value: value }
       else
         @servers.each { |s| s.unlock(resource, value) }
-        false
+
+        if errors.size >= @quorum
+          raise RedisTooManyFailuresError, "Too many Redis failures prevented lock acquisition: #{errors}"
```

Comment:
inlining errors in a string may not be very useful, how about adding it as an attribute of the exception?
```suggestion
          raise RedisTooManyFailuresError, "Too many Redis errors prevented lock acquisition", errors
```